payments-core/src/main/java/com/stripe/android/payments/core/injection/AuthenticationModule.kt

```diff
@@ -1,40 +1,83 @@
 import com.stripe.android.model.StripeIntent
 import com.stripe.android.model.StripeIntent.NextActionData
+import com.stripe.android.payments.DefaultReturnUrl
+import com.stripe.android.payments.core.authentication.DefaultPaymentAuthenticatorRegistry
 import com.stripe.android.payments.core.authentication.OxxoAuthenticator
 import com.stripe.android.payments.core.authentication.PaymentAuthenticator
 import com.stripe.android.payments.core.authentication.WebIntentAuthenticator
+import com.stripe.android.view.AuthActivityStarterHost
 import dagger.Binds
+import dagger.Lazy
 import dagger.Module
+import dagger.Provides
 import dagger.multibindings.IntoMap
+import javax.inject.Singleton
 
 /**
  * Provides mappings between [NextActionData] and [PaymentAuthenticator] provided by payment SDK.
  */
 @Module
-internal interface AuthenticationModule {
+internal abstract class AuthenticationModule {
     @IntentAuthenticatorMap
     @Binds
     @IntoMap
     @IntentAuthenticatorKey(NextActionData.SdkData.Use3DS1::class)
-    fun binds3DS1Authenticator(webIntentAuthenticator: WebIntentAuthenticator): PaymentAuthenticator<StripeIntent>
+    abstract fun binds3DS1Authenticator(webIntentAuthenticator: WebIntentAuthenticator): PaymentAuthenticator<StripeIntent>
 
     @IntentAuthenticatorMap
     @Binds
     @IntoMap
     @IntentAuthenticatorKey(NextActionData.RedirectToUrl::class)
-    fun bindsRedirectAuthenticator(webIntentAuthenticator: WebIntentAuthenticator): PaymentAuthenticator<StripeIntent>
+    abstract fun bindsRedirectAuthenticator(webIntentAuthenticator: WebIntentAuthenticator): PaymentAuthenticator<StripeIntent>
 
     @IntentAuthenticatorMap
     @Binds
     @IntoMap
     @IntentAuthenticatorKey(NextActionData.AlipayRedirect::class)
-    fun bindsAlipayRedirectAuthenticator(webIntentAuthenticator: WebIntentAuthenticator): PaymentAuthenticator<StripeIntent>
+    abstract fun bindsAlipayRedirectAuthenticator(webIntentAuthenticator: WebIntentAuthenticator): PaymentAuthenticator<StripeIntent>
 
     @IntentAuthenticatorMap
     @Binds
     @IntoMap
     @IntentAuthenticatorKey(NextActionData.DisplayOxxoDetails::class)
-    fun bindsOxxoAuthenticator(oxxoAuthenticator: OxxoAuthenticator): PaymentAuthenticator<StripeIntent>
+    abstract fun bindsOxxoAuthenticator(oxxoAuthenticator: OxxoAuthenticator): PaymentAuthenticator<StripeIntent>
+
+    companion object {
+        @Provides
+        @Singleton
+        fun provideDefaultReturnUrl(
+            context: Context
+        ) = DefaultReturnUrl.create(context)
+
+        @Provides
+        @Singleton
+        fun providePaymentRelayStarterFactory(
+            lazyRegistry: Lazy<DefaultPaymentAuthenticatorRegistry>,
+        ): (AuthActivityStarterHost) -> PaymentRelayStarter =
+            { host: AuthActivityStarterHost ->
+                lazyRegistry.get().paymentRelayLauncher?.let {
+                    PaymentRelayStarter.Modern(it)
+                } ?: PaymentRelayStarter.Legacy(host)
```

Comment:
Can we deprecate this Legacy method, so one day in the future we can delete it?

Reply:
Can you create a separate ticket to track that migration? Since `onActivityResult` is already deprecated, we should deprecate our APIs that still make use of it.

Reply:
Created [this](https://jira.corp.stripe.com/browse/MOBILESDK-353) ticket to track.